src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint_legacy.py

```diff
@@ -562,6 +567,9 @@ def __call__(
                     if callback is not None and i % callback_steps == 0:
                         callback(i, t, latents)
 
+        # use original latents corresponding to unmasked portions of the image
+        latents = (init_latents_orig * mask) + (latents * (1 - mask))
```

Comment:
Hmm this changes the behavior quite a bit of the legacy pipeline, but it does make sense! @patil-suraj @pcuenca wdyt?

Reply:
Makes sense!